src/lib.rs

```diff
@@ -192,21 +185,44 @@ pub fn build(config: &Config) -> Result<(), Box<std::error::Error>> {
 
     document.data = Some(PrimaryData::Single(Box::new(krate)));
 
-    let serialized = serde_json::to_string(&document)?;
+    Ok(document)
+}
 
-    let mut file = File::create(json_path)?;
-    file.write_all(serialized.as_bytes())?;
+pub fn build(config: &Config, artifacts: &[&str]) -> Result<(), Box<std::error::Error>> {
+    let output_path = config.manifest_path.join("target/doc");
+    fs::create_dir_all(&output_path)?;
 
-    // now that we've written out the data, we can write out the rest of it
-    let mut assets_path = output_path.clone();
-    assets_path.push("assets");
-    fs::create_dir_all(&assets_path)?;
+    if artifacts.contains(&"json") {
+        print!("generating JSON...");
+        io::stdout().flush()?;
 
-    for asset in &config.assets {
-        create_asset_file(asset.name, &output_path, asset.contents)?;
+        let document = generate_json(&config)?;
+        let serialized = serde_json::to_string(&document)?;
+
+
+        let mut json_path = output_path.clone();
+        json_path.push("data.json");
+
+        let mut file = File::create(json_path)?;
+        file.write_all(serialized.as_bytes())?;
+        println!("done.");
```

Comment:
You used `io::stdout().flush()` earlier, would you want to use it here as well? Since you call it a few times it might even be worth it to do something like:

```rust
let stdout = io::stdout();
```
then after printing or writing just call `stdout.flush()?` rather than acquiring a new handle to `stdout` each time